pkg/api/validation/schema.go

```diff
@@ -66,6 +67,57 @@ func NewSwaggerSchemaFromBytes(data []byte) (Schema, error) {
 	return schema, nil
 }
 
+// validateList unpack a list and validate every item in the list.
+// It return nil if every item is ok.
+// Otherwise it return an error list contain errors of every item.
+func (s *SwaggerSchema) validateList(obj map[string]interface{}) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	items, exists := obj["items"]
+	if !exists {
+		return append(allErrs, fmt.Errorf("no items field in %#v", obj))
+	}
+	if _, ok := items.([]interface{}); !ok {
+		return append(allErrs, fmt.Errorf("items isn't a slice"))
+	}
+	for i, item := range items.([]interface{}) {
+		fields, ok := item.(map[string]interface{})
+		if !ok {
+			allErrs = append(allErrs, fmt.Errorf("items[%d] isn't a map[string]interface{}", i))
+			continue
+		}
+		groupVersion := fields["apiVersion"]
+		if groupVersion == nil {
+			allErrs = append(allErrs, fmt.Errorf("items[%d].apiVersion not set", i))
+			continue
+		}
+		if _, ok := groupVersion.(string); !ok {
```

Comment:
Casting all over the place is not advised, instead cast once and use the result everywhere else.
```go
gv, ok := groupVersion.(string)
if !ok { /* .. */ }
// use gv 
```
Same with `kind`.